python/pyarrow/array.pxi

```diff
@@ -855,7 +858,7 @@ cdef class StructArray(Array):
         cdef DataType struct_type = struct([
             field(name, array.type)
             for name, array in
-            zip(field_names, arrays)
+            zip(names, arrays)
```

Comment:
If `names` is `None` this will fail. What's the desired behavior in the case that `names` is `None`?